src/tuning/layout_tuner.cpp

```diff
@@ -142,12 +106,26 @@ void LayoutTuner::UpdateDefaultPartition(storage::DataTable* table) {
   table->ClearLayoutSamples();
 
   // Desired number of tiles
-  auto layout = clusterer.GetPartitioning(tile_count);
-
-  LOG_TRACE("%s", GetColumnMapInfo(layout).c_str());
+  auto column_map = clusterer.GetPartitioning(tile_count);
 
+  auto table_oid = table->GetOid();
+  auto database_oid = table->GetDatabaseOid();
   // Update table layout
-  table->SetDefaultLayout(layout);
+  // Since we need the layout to be updated in the catalog,
+  // Start a transaction before updating the default layout.
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto *txn = txn_manager.BeginTransaction();
+  auto catalog = catalog::Catalog::GetInstance();
+  if (catalog->CreateDefaultLayout(database_oid, table_oid, column_map, txn) ==
+      nullptr) {
+    txn_manager.AbortTransaction(txn);
+    LOG_DEBUG("Layout Update to failed.");
```

Comment:
Actually, this is going to be a part of the `brain` code to `Tune` the table layout. A transaction may fail because it conflicts with other concurrent updates. In an ideal case it must fail because tuning is a background task while the other transaction would be user specified. Since this is a background process, in case of failure, it would come around in the next iteration and attempt the change again. 
I am not sure if it would be appropriate to throw an exception when the failure was due to an internal event rather than a user action. What do you think?

Reply:
Ok. I think it is fine not to throw an exception. 
This is not your issue, but I'm wondering why the tuner doesn't apply a bit more intelligence. Just re-trying an operation without keeping any stats or ensuring you don't end up re-trying it forever, seems optimistic.

Reply:
I will change this function to return a `bool` to indicate the success. Later, @malin1993ml can determine how to handle failures in the `Tuner` as a part of the`brain`. Can you please review the rest of the changes? This needs to be merged in order to support schema changes. 

Also the `query_logger_test` failure is the same as #1307. So I am assuming we can ignore it.

Reply:
@pervazea I have changed that part. Please let me know if there are any other changes I need to make.